src/arrayMethodSort.js

```diff
@@ -4,8 +4,28 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1; j++) {
+        if (compareFunction !== undefined) {
+          if (compareFunction(String(this[j]), String(this[j + 1])) > 0) {
+            const firstElement = this[j];
+
+            this[j] = this[j + 1];
+            this[j + 1] = firstElement;
+          }
+        } else {
+          if (String(this[j]) > String(this[j + 1])) {
+            const firstElement = this[j];
+
+            this[j] = this[j + 1];
+            this[j + 1] = firstElement;
+          }
+        }
+      }
+    }
+
+    return this;
```

Comment:
1) u are duplicating code.
make an additional function that will run when compareFunction is not given to u
2) Also, I think u should use a `do...while` loop to keep the loop running until the entire array is sorted
```suggestion
  [].__proto__.sort2 = function (compareFunction = additionalCompareFunction) {
    for (let i = 0; i < this.length - 1; i++) {
      for (let j = 0; j < this.length - 1; j++) {
        if (compareFunction(String(this[j]), String(this[j + 1])) > 0) {
          const firstElement = this[j];

          this[j] = this[j + 1];
          this[j + 1] = firstElement;
        }
      }
    }
    
    function additionalCompareFunction = (a, b) {
      ........
    }
```